### ISSUE TITLE
Request: Allow JsDoc on the options of a string literal type.

### ISSUE THREAD

**Issue by blendsdk** [Microsoft/TypeScript#25499]
Would it be possible to support JsDoc comments on the options of a string literal type, so we can leverage the comments in our code editors? (VSCode for example):

For example:
```typescript
type TMaterialIcons =
    /**
     * ![access-point-network](https://raw.githubusercontent.com/Templarian/MaterialDesign/master/icons/svg/access-point-network.svg?sanitize=true)
     */
    | 'access-point-network'
    /**
     * ![access-point](https://raw.githubusercontent.com/Templarian/MaterialDesign/master/icons/svg/account.svg?sanitize=true)
     */
    | 'account';
```

Checking the code above with an [AST explorer](https://astexplorer.net/) shows that the comments on top of the string literals are ignored. Only 'account' and  'account-point-network' are recognised as `StringLiteral`. The comments above each value don't show up in the AST explorer so there is no way for a language service provider to show them in the IntelliSense. 

Any thoughts?

**Comment by DanielRosenwasser** [comment]
Well, the compiler considers any given string literal type to be the same, and none of them have declarations per-se. So if you used a literal type in two places, it's not 100% certain you always want to show that doc comment.

**Comment by ajafff** [comment]
Using a (const) string enum should work because each member has a declaration

**Comment by blendsdk** [comment]
@DanielRosenwasser @ajafff  Yes, I understand, I didn't think of it that way. Perhaps in this case an `enum` would have been better if I didn't have 2000+ enum options. I was hoping to have cheap way to show the icon images in the IntelliSense and not having to load a large `enum` in the browser. In my case the `enum`  or the `type TMaterialIcons ...` are auto-generated from large list of options.

**Comment by mhegazy** [comment]
Related discussion in https://github.com/Microsoft/TypeScript/issues/25090

**Comment by blendsdk** [comment]
@DanielRosenwasser @mhegazy It took some imagination and a good night sleep, but I found a solution to avoid creating a large enum and show the icon images within intelliSense.

Following is a POC. 

![image](https://user-images.githubusercontent.com/5524595/42521767-e40207b2-8469-11e8-86b0-96c611ff88ed.png)
